### PR TITLE
Ssl support for the mqtt ingestion source

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: blackbox/docs/administration/ingestion/mqtt.txt [hunk 1]
```diff
@@ -60,6 +60,20 @@ Node Settings
   This timeout is used if the client does not specify a ``keepAlive`` option
   when sending the ``CONNECT`` message.
 
+SSL Support
+...........
+
+SSL support is available for the MQTT connections and can be enabled using the 
+``ssl.ingestion.mqtt.enabled`` configuration. When SSL is enabled for the MQTT
```
Comment (matriv) [inline blackbox/docs/administration/ingestion/mqtt.txt]: configuration -> setting

File: blackbox/docs/administration/ssl.txt [hunk 2]
```diff
@@ -179,6 +183,12 @@ information including the different SSL modes please visit the
 .. _psql documentation: https://www.postgresql.org/docs/current/static/app-psql.html
 
 
+Connecting to a CrateDB MQTT Broker With SSL/TLS
+------------------------------------------------
+A CrateDB node configured to run as :ref:`MQTT broker <ingest_mqtt>` can 
+accept only encrypted connections if ssl ingestion is enabled, namely if the 
+``ssl.ingestion.mqtt.enabled`` configuration is set to ``true`` and the
```
Comment (matriv) [inline blackbox/docs/administration/ssl.txt]: configuration -> setting

File: core/src/main/java/io/crate/plugin/EnterpriseLoader.java [hunk 3]
```diff
@@ -47,4 +47,5 @@
         }
         return instance;
     }
+
```
Comment (matriv) [inline core/src/main/java/io/crate/plugin/EnterpriseLoader.java]: Oh no!!!!!!! :-)

File: http/src/main/java/io/crate/plugin/PipelineRegistry.java [hunk 4]
```diff
@@ -88,22 +89,21 @@ public void addBefore(ChannelPipelineItem item) {
         }
     }
 
-    public void registerSslContext(SslContext sslContext) {
-        this.sslContext = sslContext;
-        if (sslContext != null) {
```
Comment (matriv) [inline http/src/main/java/io/crate/plugin/PipelineRegistry.java]: Would be nice to keep this logging during startup.

File: blackbox/docs/administration/ingestion/mqtt.txt [hunk 5]
```diff
@@ -60,6 +60,20 @@ Node Settings
...
   when sending the ``CONNECT`` message.
 
+SSL Support
+...........
+
+SSL support is available for the MQTT connections and can be enabled using the 
+``ssl.ingestion.mqtt.enabled`` setting. When SSL is enabled for the MQTT 
+connection, consider changing ``ingestion.mqtt.port`` to ``8883`` which is 
+the TCP/IP registered port for MQTT over SSL. 
+Once CrateDB is configured to use encrypted MQTT connections, unencrypted
```
Comment (nslater) [inline blackbox/docs/administration/ingestion/mqtt.txt]: black line above this

File: blackbox/docs/administration/ingestion/mqtt.txt [hunk 6]
```diff
@@ -60,6 +60,20 @@ Node Settings
...
+SSL support is available for the MQTT connections and can be enabled using the 
+``ssl.ingestion.mqtt.enabled`` setting. When SSL is enabled for the MQTT 
+connection, consider changing ``ingestion.mqtt.port`` to ``8883`` which is 
+the TCP/IP registered port for MQTT over SSL. 
+Once CrateDB is configured to use encrypted MQTT connections, unencrypted 
+connections will not be supported anymore.
+
+.. SEEALSO::
+
+  :doc:`/administration/ssl`
```
Comment (nslater) [inline blackbox/docs/administration/ingestion/mqtt.txt]: use `:ref:` not `:doc:` please

File: blackbox/docs/administration/ssl.txt [hunk 7]
```diff
@@ -9,14 +9,17 @@ Secured Communications (SSL/TLS)
    *Secured Communications* is an Enterprise Edition feature.
 
 Secured communication allows you to encrypt traffic between the CrateDB node
-and a client. This applies to connections using HTTP (i.e. Admin UI, Crash, Rest)
-as well as connections using the PostgreSQL Wire Protocol (i.e. JDBC, psql).
+and a client. This applies to connections using HTTP (i.e. Admin UI, Crash, Rest),
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: "REST"

link admin ui, link crash, link rest

(link them to the appropriate page in the guide)

File: blackbox/docs/administration/ssl.txt [hunk 8]
```diff
@@ -9,14 +9,17 @@ Secured Communications (SSL/TLS)
...
+the PostgreSQL Wire Protocol (i.e. JDBC, psql), and MQTT (i.e. :ref:`MQTT
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: link postgrs wire protocol

link mqtt and remove the parenthesis

File: blackbox/docs/administration/ssl.txt [hunk 9]
```diff
@@ -9,14 +9,17 @@ Secured Communications (SSL/TLS)
...
+ingestion source <ingest_mqtt>`)
+
 Connections are secured using Transport Layer Security (TLS).
 
 Note that once SSL is enabled for HTTP connections, only connections using
-HTTPS are allowed. This is in contrast to the PostgreSQL Wire Protocol which
-still allows non-encrypted connections when SSL is enabled. If you want to
-enforce SSL usage, please have a look at :ref:`administration_hba`.
+HTTPS are allowed. Same applies to secure MQTT connections.
+This is in contrast to the PostgreSQL Wire Protocol which still allows
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: new line above

Comment (nslater) [reply blackbox/docs/administration/ssl.txt]: ", which still"

File: blackbox/docs/administration/ssl.txt [hunk 10]
```diff
@@ -9,14 +9,17 @@ Secured Communications (SSL/TLS)
...
 Connections are secured using Transport Layer Security (TLS).
 
 Note that once SSL is enabled for HTTP connections, only connections using
-HTTPS are allowed. This is in contrast to the PostgreSQL Wire Protocol which
-still allows non-encrypted connections when SSL is enabled. If you want to
-enforce SSL usage, please have a look at :ref:`administration_hba`.
+HTTPS are allowed. Same applies to secure MQTT connections.
+This is in contrast to the PostgreSQL Wire Protocol which still allows 
+non-encrypted connections when SSL is enabled. If you want to enforce SSL 
+usage, please have a look at :ref:`administration_hba`.
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: "please have a look at" -> "consult"

File: blackbox/docs/administration/ssl.txt [hunk 11]
```diff
@@ -179,6 +183,12 @@ information including the different SSL modes please visit the
 .. _psql documentation: https://www.postgresql.org/docs/current/static/app-psql.html
 
 
+Connecting to a CrateDB MQTT Broker With SSL/TLS
+------------------------------------------------
+A CrateDB node configured to run as :ref:`MQTT broker <ingest_mqtt>` can
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: "as an"

File: blackbox/docs/administration/ssl.txt [hunk 12]
```diff
@@ -179,6 +183,12 @@ information including the different SSL modes please visit the
 .. _psql documentation: https://www.postgresql.org/docs/current/static/app-psql.html
 
 
+Connecting to a CrateDB MQTT Broker With SSL/TLS
+------------------------------------------------
+A CrateDB node configured to run as :ref:`MQTT broker <ingest_mqtt>` can 
+accept only encrypted connections if ssl ingestion is enabled, namely if the
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: "SSL"

Comment (nslater) [reply blackbox/docs/administration/ssl.txt]: "can only accept"

File: blackbox/docs/administration/ssl.txt [hunk 13]
```diff
@@ -179,6 +183,12 @@ information including the different SSL modes please visit the
 .. _psql documentation: https://www.postgresql.org/docs/current/static/app-psql.html
 
 
+Connecting to a CrateDB MQTT Broker With SSL/TLS
+------------------------------------------------
+A CrateDB node configured to run as :ref:`MQTT broker <ingest_mqtt>` can 
+accept only encrypted connections if ssl ingestion is enabled, namely if the 
+``ssl.ingestion.mqtt.enabled`` setting is configured to ``true`` and the
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: link the setting

File: blackbox/docs/administration/ssl.txt [hunk 14]
```diff
@@ -179,6 +183,12 @@ information including the different SSL modes please visit the
 .. _psql documentation: https://www.postgresql.org/docs/current/static/app-psql.html
 
 
+Connecting to a CrateDB MQTT Broker With SSL/TLS
+------------------------------------------------
+A CrateDB node configured to run as :ref:`MQTT broker <ingest_mqtt>` can 
+accept only encrypted connections if ssl ingestion is enabled, namely if the 
+``ssl.ingestion.mqtt.enabled`` setting is configured to ``true`` and the 
+``ingestion.mqtt.port`` setting is ``8883``.
```
Comment (nslater) [inline blackbox/docs/administration/ssl.txt]: link the setting

**Pull request merged**